### PR TITLE
LAGRANGE: Add instrument definition

### PR DESCRIPTION
**Pull request by darominski**
which is used only to allow for distinction between data spectrum and the monitor one.

**Description of work.**

This PR adds a mockup instrument for Lagrange instrument. The dimensions or positions of the detector is not relevant in the reduction workflow. However, a way to properly distinguish data from monitor spectra requires an instrument, as well as to profit from parameters that can be stored in an Instrument Parameter File.

**To test:**
1. In workbench, run `LoadEmptyInstrument` with `InstrumentName` set to Lagrange.
2. Show data, it should have two spectra, with the second being a monitor.

Part of #35152  

*This does not require release notes* because ** the full release note of adding Lagrange will be added when the NeXus loader is implemented and tested.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: instrument/Lagrange_Definition.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<?xml version='1.0' encoding='ASCII'?>
+<instrument xmlns="http://www.mantidproject.org/IDF/1.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="Lagrange" valid-from="1901-01-01 00:00:00" valid-to="2100-01-31 23:59:59" last-modified="2023-02-06 14:34:26.766171" xsi:schemaLocation="http://www.mantidproject.org/IDF/1.0 http://schema.mantidproject.org/IDF/1.0/IDFSchema.xsd">
+  <!--
+       This is the instrument definition file of the 
+       Generated file, PLEASE DO NOT EDIT THIS FILE!
+       This file was automatically generated by mantidgeometry/ILL/IDF/salsa_generateIDF.py
+
+       For more information, please visit
+       https://www.ill.eu/instruments-support/instruments-groups/instruments/salsa/characteristics/
+       -->
```
Comment (gui-co) [inline instrument/Lagrange_Definition.xml:10]: I see that you had the inspiration from Salsa :)

**Review by gui-co — CHANGES_REQUESTED** [review]
The comment needs to be updated.

Because of the detector size (1*1), the instrument can not be displayed in "Full 3D" in the instrument view (`Cannot display rectangular detector bank detector
Detector step in x direction is zero.`). It is maybe not a problem as the instrument is not really used during the reduction.

File: instrument/Lagrange_Definition.xml (hunk 1 above)
Comment (darominski) [reply instrument/Lagrange_Definition.xml:10]: Yes indeed, shamelessly copied methods from both SALSA and D11.

**Comment by darominski** [comment]
> The comment needs to be updated.
> 
> Because of the detector size (1*1), the instrument can not be displayed in "Full 3D" in the instrument view (`Cannot display rectangular detector bank detector Detector step in x direction is zero.`). It is maybe not a problem as the instrument is not really used during the reduction.

The instrument is not used in the reduction and the visualisation is not planned to be used at all. The instrument is added so that instrument parameter file can be used and for proper flagging of the monitor spectrum.

File: instrument/Lagrange_Definition.xml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+<?xml version='1.0' encoding='ASCII'?>
+<instrument xmlns="http://www.mantidproject.org/IDF/1.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="Lagrange" valid-from="1901-01-01 00:00:00" valid-to="2100-01-31 23:59:59" last-modified="2023-02-09 11:35:48.875671" xsi:schemaLocation="http://www.mantidproject.org/IDF/1.0 http://schema.mantidproject.org/IDF/1.0/IDFSchema.xsd">
+  <!--
+       This is the instrument definition file of the 
+       Generated file, PLEASE DO NOT EDIT THIS FILE!
+       This file was automatically generated by mantidgeometry/ILL/IDF/lagrange_generateIDF.py
+
+       For more information, please visit
+       https://www.ill.eu/instruments-support/instruments-groups/instruments/lagrange/characteristics/
```
Comment (gui-co) [inline instrument/Lagrange_Definition.xml:9]:
```suggestion
       https://www.ill.eu/users/instruments/instruments-list/in1-taslagrange/characteristics
```

Comment (gui-co) [reply instrument/Lagrange_Definition.xml:9]: The link is not working. This one should be fine.

**Review by gui-co — CHANGES_REQUESTED** [review]

**Review by gui-co — APPROVED** [review]
Thanks for the changes. Everything looks fine.

**Review by eurydice76 — APPROVED** [review]
The XML file looks to us.

**Pull request merged**